language/stdlib/modules/DesignatedDealer.move

```diff
@@ -225,12 +280,34 @@ module DesignatedDealer {
                 dealer.window_inflow == old(dealer.window_inflow) + amount);
         ensures tier_index < len(old(dealer).tiers);
         ensures dealer.window_inflow <= old(dealer).tiers[tier_index];
+
+        include TieredMint<CoinType>;
+    }
+
+    spec schema TieredMint<CoinType> {
+        tc_account: signer;
+        dd_addr: address;
+        amount: u64;
+        tier_index: u64;
+        include Roles::AbortsIfNotTreasuryCompliance{account: tc_account};
+        aborts_if amount == 0 with Errors::INVALID_ARGUMENT;
+        include AbortsIfDealerNotExistAt;
+        aborts_if !exists<TierInfo<CoinType>>(dd_addr);
```

Comment:
Missing aborts code.

Reply:
This aborts condition is supposed to cover the unguarded borrow_mut of TierInfo<CoinType>(dd_addr).  If we want to instead use something like the following schema:

   spec schema AbortsIfNoTierInfo<CoinType> {
        dd_addr: address;
        aborts_if !exists<TierInfo<CoinType>>(dd_addr) with Errors::NOT_PUBLISHED;
    }

then we have to change the code and add an explicit assert before the borrow_mut.

What is the recommendation here, @tzakian , @wrwg ?

Reply:
I think adding the assertion check makes sense here since we want to remove any possibility of hitting an `ExecutionFailure`.

Reply:
I have added the assertion check.